test/nightly/run_combined_tests.py

```diff
@@ -0,0 +1,464 @@
+        return False, error_msg, None
+
+    finally:
+        if process:
+            kill_process_tree(process.pid)
+
+
+def run_metrics(
+    models: List[ModelLaunchSettings],
+    run_perf: bool = True,
+    run_accuracy: bool = True,
+    is_vlm: bool = False,
+    base_url: Optional[str] = None,
+    profile_dir: Optional[str] = None,
+    test_name: str = "NightlyMetrics",
+    # Performance test parameters
+    batch_sizes: Optional[List[int]] = None,
+    input_lens: Optional[Tuple[int, ...]] = None,
+    output_lens: Optional[Tuple[int, ...]] = None,
+    # Accuracy test parameters
+    eval_name: Optional[str] = None,
+    num_examples: Optional[int] = None,
+    num_threads: int = 1024,
+    max_tokens: Optional[int] = None,
+    # VLM-specific parameters
+    dataset_name: Optional[str] = None,
+    return_latency: bool = False,
+    # Additional eval kwargs
+    **eval_kwargs,
+) -> dict:
+    """Run performance and/or accuracy tests for a list of models.
+
+    This function uses the same abstractions as existing nightly tests:
+    - NightlyBenchmarkRunner for performance tests
+    - run_eval (SimpleNamespace) for accuracy tests
+
+    Args:
+        models: List of ModelLaunchSettings to test
+        run_perf: Whether to run performance tests
+        run_accuracy: Whether to run accuracy tests
+        is_vlm: Whether these are VLM models (affects defaults)
+        base_url: Server base URL (default: DEFAULT_URL_FOR_TEST)
+        profile_dir: Directory for performance profiles
+        test_name: Name for the test (used in reports)
+
+        # Performance test parameters:
+        batch_sizes: Batch sizes for perf test (default: [1, 1, 8, 16, 64])
+        input_lens: Input lengths (default: from NIGHTLY_INPUT_LENS env or "4096")
+        output_lens: Output lengths (default: from NIGHTLY_OUTPUT_LENS env or "512")
+
+        # Accuracy test parameters:
+        eval_name: Evaluation name (default: "mgsm_en" for text, "mmmu" for VLM)
+        num_examples: Number of examples (default: None for text, 100 for VLM)
+        num_threads: Number of threads for evaluation (default: 1024)
+        max_tokens: Max tokens for generation
+
+        # VLM-specific:
+        dataset_name: Dataset name for VLM benchmarks (default: "mmmu")
+        return_latency: Whether to return latency metrics for VLM eval
+
+        # Additional eval parameters:
+        **eval_kwargs: Any additional parameters to pass to run_eval
+                       (e.g., thinking_mode, temperature, repeat)
+
+    Returns:
+        dict with test results:
+        {
+            "all_passed": bool,
+            "results": [
+                {
+                    "model": str,
+                    "perf_passed": bool/None,
+                    "accuracy_passed": bool/None,
+                    "accuracy_metrics": dict/None,
+                    "errors": list,
+                },
+                ...
+            ]
+        }
+
+    Example:
```

Comment:
Let's remove the long comment as people can look at model test file as an example